### PR TITLE
Compute MD5s in dialyzer using the .beam file

### PR DESCRIPTION
**Pull request by josevalim**
The previous mechanism was based on the core file which
meant that for every module in the PLT, we had to fetch
its .beam file, retrieve Erlang AST, compile that down
to core, serialize it into a binary and then get its MD5.

In a project with stdlib, kernel, elixir and a small
application in the PLT,  relying on beam_lib sped up
--check_plt from 10s to 0.8s.

Other workflows, such as checking and adding new modules
to a PLT, have also seen improvements in the range of 2x
to 3x.

### CODE REVIEW

**Comment by uabboli** [comment]
Can your patch handle changes of included files?

**Comment by josevalim** [comment]
@uabboli good catch, it cannot. We can solve this by getting all chunks and discarding only the ones that do not affect Dialyzer: "CInf", "Docs" and "Attr". I will update it shortly.

**Comment by josevalim** [comment]
I have force pushed and updated the issue description.

**Comment by josevalim** [comment]
The build failed because the spec for beam_lib:all_chunks was wrong. I have pushed a new commit but I am not sure if you want me to squash them together since they may be considered unrelated. Your call. :)

File: lib/dialyzer/src/dialyzer_plt.erl [hunk 1]
```diff
@@ -531,17 +531,19 @@ compute_md5_from_files(Files) ->
...
 
 compute_md5_from_file(File) ->
-  case filelib:is_regular(File) of
-    false ->
+  case beam_lib:all_chunks(File) of
+    {ok, _, Chunks} ->
+      %% We cannot use beam_lib:md5 because it does not consider
+      %% the debug_info chunk, where typespecs are likely stored.
+      %% So we consider almost all chunks except the useless ones.
+      Filtered = [[ID, Chunk] || {ID, Chunk} <- Chunks, ID /= "CInf", ID /= "Docs"],
```
Comment (kostis) [inline lib/dialyzer/src/dialyzer_plt.erl]: I suggest to use exact comparison operators here.

Comment (uabboli) [reply lib/dialyzer/src/dialyzer_plt.erl]: Keep two commits, please.

**Comment by josevalim** [comment]
Ok, all comments addressed. I have also changed the commit order so we don't have an execution flow where the build fails.

**Pull request merged**